Sources/MapboxMaps/Style/StyleProjection.swift

```diff
@@ -0,0 +1,44 @@
+/// Describes the projection used to render the map.
+public struct StyleProjection: Hashable, Codable {
+
+    /// The name of the projection.
+    public var name: StyleProjectionName
+
+    /// Initializes a projection
+    public init(name: StyleProjectionName) {
+        self.name = name
+    }
+
+    internal enum CodingKeys: String, CodingKey {
+        case name
+    }
+}
+
+/// The name of a style projection as defined in the Mapbox Style Spec
+public struct StyleProjectionName: RawRepresentable, Hashable, Codable {
```

Comment:
Is there a reason we are using struct instead of enum here?

Reply:
struct should be easier to extend without breaking changes. adding a case to an non-frozen-enum is source-breaking.